src/Symfony/Component/Form/Extension/Core/Type/DateIntervalType.php

```diff
@@ -192,6 +195,21 @@ public function configureOptions(OptionsResolver $resolver)
             return array_fill_keys($timeParts, $placeholder);
         };
 
+        $labelsNormalizer = function (Options $options, array $labels) {
+            return array_replace(array(
+                'years' => null,
+                'months' => null,
+                'days' => null,
+                'weeks' => null,
+                'hours' => null,
+                'minutes' => null,
+                'seconds' => null,
+                'invert' => 'Negative interval',
+            ), array_filter($labels, function ($label) {
```

Comment:
I prefer your approach :). Problem is that it mostly depends on the widget we consider.

Having the possibility to define labels is great, mainly to translate them. It means we should be able to define all of them (and that would be consistent).

Also regarding performances, your approach should be better, using one normalizer for one array (even calling an `array_replace` and an `array_filter`) instead of defining many options. Resolving only one option involves cloning the resolver instance and many calls, so that must cost much more.

In conclusion, I'd say maybe #20002 should follow what's implemented here, using `date_label` and `time_label` as string or array, so maybe `DateType` and `TimeType` should have a `labels` option too.

What do you and others think?